### PR TITLE
fix turbomind CUDA runtime error invalid argument

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

https://github.com/InternLM/lmdeploy/issues/1058

## Modification

workaround by distinguishing between different backends

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
Hi @lvhan028 @AllentDan @lzhangzz May you help review this pr? Thanks.

**Comment by zhyncs** [comment]
cc @grimoire

File: lmdeploy/serve/async_engine.py [hunk 1]
```diff
@@ -247,19 +247,28 @@ def safe_run(self, session_id: Optional[int] = None):
 
     async def get_generator(self, stop: bool, session_id: int):
         """Only return the model instance if it is available."""
-        if stop:
-            return self.engine.create_instance()
-        while self.gens_set == set():
-            await asyncio.sleep(0)
-        if str(session_id) in self.id2generator:
```
Comment (AllentDan) [inline lmdeploy/serve/async_engine.py:254]: Just append `and self.backend == 'pytorch` please.

Comment (zhyncs) [reply lmdeploy/serve/async_engine.py:254]: > Just append `and self.backend == 'pytorch` please.

ok

Comment (zhyncs) [reply lmdeploy/serve/async_engine.py:254]: > Just append `and self.backend == 'pytorch` please.

done

**Review by AllentDan — APPROVED** [review]

**Comment by zhyncs** [comment]
fix https://github.com/InternLM/lmdeploy/issues/1058

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**